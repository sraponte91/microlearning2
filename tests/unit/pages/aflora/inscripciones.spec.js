import { config, createLocalVue, mount } from '@vue/test-utils'
import VueMeta from 'vue-meta'
import AfloraInscripciones from '@/pages/aflora/inscripciones.vue'
// create vue with vue-meta
const localVue = createLocalVue()
localVue.use(VueMeta, { keyName: 'head' })

const mock = {
    localVue,
    mocks: {}
}

const wrapper = mount(AfloraInscripciones, mock)

describe('inscripciones.vue', () => {
    test('Is a Vue instance', () => {
        expect(wrapper.vm).toBeTruthy()
    })
    test('Renders correctly', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
    test('has correct <head> content', () => {
        expect(wrapper.vm.$metaInfo.title)
            .toBe(config.mocks['$afloraData'].afloraInscripciones.meta.title +
                ' - ' + config.mocks['$constants'].home.meta.title)

        expect(wrapper.vm.$metaInfo.meta.find(
            (item) => item.hid === 'title'
        ).content)
            .toBe(config.mocks['$afloraData'].afloraInscripciones.meta.metaTitle +
                ' - ' + config.mocks['$constants'].home.meta.title)

        expect(wrapper.vm.$metaInfo.meta.find(
            (item) => item.hid === 'description'
        ).content).toBe(config.mocks['$afloraData'].afloraInscripciones.meta.metaDescription)

        expect(wrapper.vm.$metaInfo.meta.find(
            (item) => item.hid === 'og:title'
        ).content)
            .toBe(config.mocks['$afloraData'].afloraInscripciones.meta.ogTitle +
                ' - ' + config.mocks['$constants'].home.meta.title)

        expect(wrapper.vm.$metaInfo.meta.find(
            (item) => item.hid === 'og:description'
        ).content).toBe(config.mocks['$afloraData'].afloraInscripciones.meta.ogDescription)

        expect(wrapper.vm.$metaInfo.meta.find(
            (item) => item.hid === 'og:image'
        ).content).toBe(config.mocks['$constants'].baseURL + config.mocks['$afloraData'].afloraInscripciones.meta.ogImage)
    })
})
