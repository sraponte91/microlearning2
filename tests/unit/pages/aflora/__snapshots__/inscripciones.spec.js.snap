// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inscripciones.vue Renders correctly 1`] = `
<main>
  <div
    class="breadc-bak"
  >
    <div
      class="container margin-0-auto"
    >
      <ol
        class="breadcrumb ff-sans-r mb-2"
      >
        <li
          class="breadcrumb-item"
        >
          <a
            class=""
            href="/"
            target="_self"
          >
            Home
          </a>
        </li>
         
        <li
          class="breadcrumb-item"
        >
          <a
            class=""
            href="/aflora"
            target="_self"
          >
            Aflora
          </a>
        </li>
         
        <li
          class="breadcrumb-item active"
        >
          <span
            aria-current="location"
          >
            Inscripciones
          </span>
        </li>
      </ol>
       
      <div
        class="back-to-home position-relative d-flex align-items-center small-size"
      >
        <i
          class="ic-back"
        />
         
        <a
          class="ff-sans-b"
          href="javascript:void(0)"
        >
          Volver
        </a>
      </div>
    </div>
  </div>
   
  <section
    class="ig-tx-zing-zag desarrolla inscripci-pg bg-white"
  >
    <div
      class="container"
    >
      <div
        class="sm-siz-container margin-0-auto"
      >
        <div
          class="d-flex multiple-reapt align-items-start flex-wrap-991"
        >
          <h2
            class="ff-sans-b ss d-block-991 d-none-n wow bounceInLeft"
            data-wow-duration="2s"
          >
            
            Inscripciones
          
          </h2>
           
          <h3
            class="ff-sans-b ss d-block-991 d-none-n wow bounceInLeft"
            data-wow-duration="2s"
          >
            
            ¿Cómo inscribirse 
            <br />
            
            a Aflora?
          
          </h3>
           
          <div
            class="txt-points position-relative order-2-991 wow bounceInLeft"
            data-wow-duration="2s"
          >
            <h2
              class="ff-sans-b d-none-991"
            >
              Inscripciones
            </h2>
             
            <h3
              class="ff-sans-b d-none-991"
            >
              
              ¿Cómo inscribirse 
              <br />
              
              a Aflora?
            
            </h3>
             
            <div
              class="pra-p for-bgcol"
            >
              <p>
                
                La inscripción a nuestro programa de fortalecimiento virtual,
                será el inicio de un sendero lleno de historias y conocimiento
                que le brindará a su organización todas las herramientas para
                mejorar y lograr los objetivos planteados.
              
              </p>
            </div>
          </div>
           
          <div
            class="img-points order-1-991 wow bounceInRight"
            data-wow-duration="2s"
          >
            <img
              alt=""
              class="object-fit-cover"
              src="/images/programa-aflora/6-inscripciones/inscripciones_bloque1.jpg"
            />
          </div>
        </div>
         
        <div
          class="txt-info-m margin-0-auto text-center"
        >
          <h4
            class="ff-sans-b"
          >
            Reglamento de inscripción
          </h4>
           
          <div
            class="pra-p ff-nunito"
          >
            <p>
              
              Para Fundación Bolívar Davivienda es muy importante que todos
              los interesados, aspirantes y beneficiarios, conozcan con
              detalle las políticas detrás de cada programa. Por eso, le
              invitamos a leer con detenimiento el reglamento del programa
              Aflora. Allí encontrará los requisitos y condiciones para ser
              parte.
            
            </p>
          </div>
           
          <div
            class="two-btn d-flex justify-content-between ff-sans-b flex-wrap"
          >
            <a
              class="btn-sm btn-line-btn text-center"
              href="/legales/Footer_Reglamento_Inscripciones_Programas.pdf"
            >
              Lee el reglamento de inscripción
            </a>
             
            <a
              class="btn-sm btn-solid-btn text-center"
              href="https://fundacionbolivardaviviendaorg.force.com/comunidadfbd/s/?language=es_CO&programa=Aflora&grupo=Estudiantes+de+escuelas"
              target="_blank"
            >
              Inscribase aquí
            </a>
          </div>
        </div>
      </div>
    </div>
  </section>
</main>
`;
